substrate/frame/opf/src/types.rs

```diff
@@ -0,0 +1,200 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Types & Imports for Distribution pallet.
+
+pub use super::*;
+
+pub use frame_support::{
+	pallet_prelude::*,
+	traits::{
+		fungible,
+		fungible::{Inspect, Mutate, MutateHold},
+		fungibles,
+		schedule::{
+			v3::{Anon as ScheduleAnon, Named as ScheduleNamed},
+			DispatchTime, MaybeHashed,
+		},
+		tokens::{Precision, Preservation},
+		Bounded, DefensiveOption, EnsureOrigin, LockIdentifier, OriginTrait, QueryPreimage,
+		StorePreimage,
+	},
+	transactional, PalletId, Serialize,
+};
+pub use pallet_conviction_voting::Conviction;
+pub use frame_system::{pallet_prelude::*, RawOrigin};
+pub use scale_info::prelude::vec::Vec;
+pub use sp_runtime::traits::{
+	AccountIdConversion, BlockNumberProvider, Convert, Dispatchable, Saturating, StaticLookup, Zero,
+};
+pub use sp_std::boxed::Box;
```

Comment:
why do we reexport all those types publicly, I think the doc would be more clear if we only export the types specific for the library.